numpy/lib/tests/test_arraysetops.py

```diff
@@ -77,6 +77,42 @@ def test_ediff1d(self):
         assert(isinstance(ediff1d(np.matrix(1)), np.matrix))
         assert(isinstance(ediff1d(np.matrix(1), to_begin=1), np.matrix))
 
+    def test_isin(self):
+        # the tests for in1d cover most of isin's behavior
+        # if in1d is removed, would need to change those tests to test
+        # isin instead.
+        def _isin_slow(a, b):
+            b = np.asarray(b).flatten().tolist()
+            return a in b
+        isin_slow = np.vectorize(_isin_slow, otypes=[bool], excluded={1})
+        def assert_isin_equal(a, b):
+            x = isin(a, b)
+            y = isin_slow(a, b)
+            assert_array_equal(x, y)
+        #multidimensional arrays in both arguments
+        a = np.arange(24).reshape([2, 3, 4])
+        b = np.array([[10, 20, 30], [0, 1, 3], [11, 22, 33]])
+        assert_isin_equal(a, b)
+        #array-likes as both arguments
+        c = [(9, 8), (7, 6)]
+        d = (9, 7)
+        assert_isin_equal(c, d)
+        #zero-d scalar as:
```

Comment:
These tests are good, but tests for 0d arrays are also important, and usually catch more bugs.

Relatedly - should `isin(1, 1)` return `np.array(True)` or `np.bool_(True)`?

Also, I think the comments labelling which arg is a scalar are a bit much. This'd be more readable with just the section headings, and some space between ssections